gov.graphql

```diff
@@ -0,0 +1,44 @@
+type Proposal {
+  id: String
+  title: String
+  ipfsHash: String
+  startTime: Int # i.e. a unix timestamp; some other time type is also ok
+  endTime: Int
+  status: ProposalStatus
+  options: [ProposalOption]
+}
+
+enum ProposalStatus {
+  DISCUSSION
+  WITHDRAWN
+  VOTING_STARTED
+  VOTING_ENDED
+}
+
+type ProposalOption {
+  id: Int
+  label: String
+  voteBalanceSum: Float # convenient for making a quick status summary
+  votes: [Vote]
+}
+
+type Vote {
+  address: Address
+  balance: Float
+}
+
+type Query {
+  # when blockNumber is specified, the votes and vote balances returned should
+  # match the state at that block. when it is not specified, it should default
```

Comment:
Just to make sure it's mentioned: when a vote has reached a `VOTING_ENDED` state, all child vote balances should be frozen for that proposal at it's ending block (& same for all of its ProposalOptions' `voteBalanceSum`).

Reply:
ah, yes, that makes more sense. so the default is the latest block or the block at which the vote ended, whichever is earlier.